packages/ui/src/machines/actors/auth/signUp.ts

```diff
@@ -167,13 +168,19 @@ export const signUpActor = createMachine<SignUpContext, AuthEvent>(
       setUser,
     },
     services: {
-      async confirmSignUp(_, event) {
-        const { username, confirmation_code: code } = event.data;
+      async confirmSignUp(context, event) {
+        const { user, authAttributes } = context;
+        const { confirmation_code: code } = event.data;
+
+        const username =
+          get(user, 'username') || get(authAttributes, 'username');
```

Comment:
Can you share more about what's going on here? I can't recall where `user.username` and `authAttributes.username` both come from for this screen.

(It may be worth a code comment so we can understand why the precedence exists)

Reply:
`user` and `authAttributes` get set in different places depending on which screen we are coming from - either `SignIn` or `SignUp`. Both screens can redirect to this `ConfirmSignUp` screen, so we need to check in a couple places for the `username` value. If the `user` value is set into the context (via `SignUp`), then we will use that, but if we are coming from `SignIn`, then we need to grab the value from the `authAttributes` instead.

Me using lodash's `get` is just preference here instead of dealing with nullish coalescense stuff.